telematic_system/historical_data_processing/rosbag2_processing_service/rosbag2_processing_service/rosbag_processor.py

```diff
@@ -0,0 +1,131 @@
+        self.influx_org = influx_org
+        self.influx_client = InfluxDBClient(url=influx_url, token=influx_token, org=influx_org)
+        self.topic_exclusion_list = topic_exclusion_list
+        self.rosbag_dir = rosbag_dir
+        self.logger = logger
+
+        #Fields in the ros message to force to string type.
+        self.to_str_fields= to_str_fields
+        # Fields in the ros message to ignore
+        self.ignore_fields= ignore_fields
+
+        # Create write API
+        self.write_api = self.influx_client.write_api(write_options=ASYNCHRONOUS)
+
+        # Processing status
+        self.is_processing = False
+
+
+    async def process_rosbag(self,rosbag2_name):
+        measurement_name = rosbag2_name.split('.mcap')[0] # Measurement name is rosbag name without mcap extension
+
+        rosbag_path = os.path.join(self.rosbag_dir, rosbag2_name)
+
+        # Load the rosbag from the config directory
+        for msg in read_ros2_messages(rosbag_path):
+            try:
+                if msg.channel.topic not in self.topic_exclusion_list:
+                    topic = msg.channel.topic
+                    ros_msg = msg.ros_msg
+                    msg_attributes = self.extract_attributes(ros_msg)
+                    msg_timestamp = msg.publish_time_ns
+
+                    record = f"{measurement_name},topic_name={topic},"
+
+                    for attr_name, attr_value in msg_attributes:
+                        if attr_name in self.ignore_fields:
+                            continue
+
+                        elif attr_name in self.to_str_fields:
+                            attr_value = f'"{attr_value}"'
+                            record += f"{attr_name}={attr_value},"
+
+                        elif isinstance(attr_value, list):  # Handle arrays
+                            record += f'{attr_name}="{str(attr_value)}",'
+                        else:
+                            if isinstance(attr_value, str):
+                                attr_value = f'"{attr_value}"'  # Correctly format string values
+                            record += f"{attr_name}={attr_value},"
+
+
+                    # Remove last comma
+                    record = record[:-1]
+                    # Add timestamp at the end
+                    record += f" timestamp={msg_timestamp}"
+                    #Write record to influx
+                    self.write_api.write(bucket=self.influx_bucket, org=self.influx_org, record=record)
+
+            except Exception as e:
+                self.logger.warn(f"Failed to process ros message with exception: " + str(e))
+        self.logger.info(f"Completed rosbag processing for {rosbag2_name}")
+        self.is_processing = False
+
+
+    def extract_attributes(self, obj, parent_attr=None):
+        attributes = []
+        for attr_name in dir(obj):
+            try:
+                if not callable(getattr(obj, attr_name)) and not attr_name.startswith("__") and not attr_name.startswith("_"):
```

Comment:
The amount of nesting in this loop can be cleaned up:
```python
for attr_name in dir(obj):
  if callable(getattr(obj, attr_name)) or attr_name.startswith("__") or attr_name.startswith("_"):
    continue

  # ...
```
Additionally, it looks like two of your conditional cases can be combined:
```python
for attr_name in dir(obj):
  if callable(getattr(obj, attr_name)) or attr_name.startswith("_"):
    continue

  # ...
```
Since starting with _ includes __.